src/ServiceControl.Monitoring/Http/Diagrams/DiagramApiModule.cs

```diff
@@ -18,36 +18,43 @@ public class MonitoredEndpointsModule : ApiModule
         /// </summary>
         public MonitoredEndpointsModule(EndpointRegistry endpointRegistry, CriticalTimeStore criticalTimeStore, ProcessingTimeStore processingTimeStore, RetriesStore retriesStore, QueueLengthStore queueLengthStore)
         {
-            Get["/monitored-endpoints"] = x =>
+            const int DefaultHistory = 5;
+
+            Get["/monitored-endpoints"] = parameters =>
             {
                 var endpoints = GetMonitoredEndpoints(endpointRegistry);
+                var period = HistoryPeriod.FromMinutes((int?)Request.Query["history"] ?? DefaultHistory);
```

Comment:
does the parameters collection also cover `Request.Query`? I think this could be `(int?)parameters.history) ?? DefaultHistory`.